nextest-runner/src/test_list.rs

```diff
@@ -86,6 +85,11 @@ impl<'g> RustTestArtifact<'g> {
                         if artifact.target.name != package.name() {
                             binary_id.push_str("::");
                             binary_id.push_str(&artifact.target.name);
+                        } else if !artifact.target.kind.contains(&"lib".to_owned()) {
+                            if let Some(kind) = artifact.target.kind.get(0) {
```

Comment:
Based on https://doc.rust-lang.org/cargo/commands/cargo-metadata.html it seems like kind will only ever have one item in it unless it's a lib. I don't have any experience working with cargo metadata, though, so I'm not sure this is the right approach.

This also means if there's nothing in the kind vec, it will silently use the package name for the binary_id - should this be an error instead?

Reply:
Nothing in the kind vec is definitely weird and should at least be a warning I think.